contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/wifi_test.py

```diff
@@ -0,0 +1,296 @@
+                "{} c add type {} ifname {} con-name {} \
+                    wifi-p2p.peer {}".format(
+                    self._command,
+                    self.type,
+                    self.interface,
+                    self.conname,
+                    self.peer,
+                )
+            )
+
+    def set_band_channel(self):
+        self.run_command(
+            "{} c modify {} wifi.band {} wifi.channel {}".format(
+                self._command, self.conname, self.band, self.channel
+            )
+        )
+
+    def set_secured(self):
+        self.run_command(
+            "{} c modify {} wifi-sec.key-mgmt {} wifi-sec.psk {}\
+                  wifi-sec.group {}".format(
+                self._command,
+                self.conname,
+                self.key_mgmt,
+                self.ssid_pwd,
+                self.group,
+            )
+        )
+
+    def get_ip_addr(self):
+        """
+        nmcli -g IP4.ADDRESS command will return the IPv4 address of the
+        interface with netmask.
+        e.g. 10.102.99.224/22
+        """
+        try:
+            ip_addr = self.run_command(
+                "{} -g IP4.ADDRESS device show {}".format(
+                    self._command,
+                    self.interface,
+                )
+            ).split("/")[0]
+            return ip_addr
+        except Exception:
+            logging.warning("Not able to get IP address!")
+            raise
+
+    def up_conn(self):
+        try:
+            self.run_command("{} c up {}".format(self._command, self.conname))
+            logging.info("Initialized connection successful!")
+        except Exception:
+            raise SystemError("Bring up connection failed!")
+        for i in range(1, 6):
+            try:
+                ip_addr = self.get_ip_addr()
+                if ip_addr:
+                    logging.info("IP address is {}".format(ip_addr))
+                    return True
+            except Exception:
+                pass
+            time.sleep(5)
+        return False
+
+    def del_conn(self):
+        self.run_command("{} c delete {}".format(self._command, self.conname))
+
+    def __enter__(self):
+        self.init_conn()
+        if self.type == "wifi":
+            self.set_band_channel()
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        logging.info("Exiting context and cleaning up connection")
+        self.del_conn()
+
+
+def sshpass_cmd_gen(ip, user, pwd, cmd):
+    return "sshpass -p {} ssh -o StrictHostKeyChecking=no {}@{} {}".format(
+        pwd, user, ip, cmd
+    )
+
+
+def ping_cmd(ip):
+    return "ping {} -c 4".format(ip)
+
+
+def connect_host_device(manager, ip, user, pwd):
+    connect_cmd = "{} d wifi c {}".format(manager._command, manager.ssid)
+    if manager.key_mgmt != "none":
+        connect_cmd += " password {}".format(manager.ssid_pwd)
+    if manager.mode == "adhoc":
+        connect_cmd += " wifi.mode {}".format(manager.mode)
```

Comment:
I would suggest we move this part into `WiFiManager` as a function, called `connect_me` or something similar.
because of the connect_cmd is related to what wireless mode we initial in the `WiFiManager`

Reply:
I've pushed the fix